### PR TITLE
Stop passing WindowsHostProcessContainer annotations for CRI calls in kubelet

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR removes annotations added to CRI pod and container creation calls for Windows HostProcess containers.
These annotations were added to enable Windows HostProcess containers ahead of containerD understanding the new CRI fields for this feature.

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/labels.go#L46-L50

https://github.com/containerd/containerd/pull/5131 added support for the CRI frields relating to HostProcess containers

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/enhancements/issues/1981

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support#cri-implementation-details
```

/sig windows
/sig node
/area kubelet
/milestone v1.23

### CODE REVIEW

File: pkg/kubelet/kuberuntime/labels.go [hunk 1]
```diff
@@ -106,15 +96,6 @@ func newPodAnnotations(pod *v1.Pod) map[string]string {
 		annotations[k] = v
```
Comment (jsturtevant) [inline pkg/kubelet/kuberuntime/labels.go:96]: This loop was added to handle this special case, should we put it back to what it was before? https://github.com/kubernetes/kubernetes/commit/fd94032b2132e740a70449ffb3e9fc8f66d2c9d4#diff-a7e9fc1284dd3aa38fe530c010397a14d657f4b81e01217344c053ad6d5454a3L92

Comment (marosset) [reply pkg/kubelet/kuberuntime/labels.go:96]: Yea, I can do that. Good catch!

**Review by mrunalp — APPROVED** [review]

**Pull request merged**